class.mesh-controls.php

```diff
@@ -0,0 +1,455 @@
+        }
+
+		return $options;
+    }
+
+    function get_offset_options( $block, $section_blocks ) {
+
+	    $default_block_columns = 12 / $section_blocks;
+	    $block_columns = get_post_meta( $block->ID, '_mesh_column_width', true );
+
+	    if ( empty( $block_columns ) ) {
+	        $block_columns = $default_block_columns;
+        }
+
+	    $offsets_available = $block_columns - 3;
+
+	    $options = array();
+
+        for ( $i = 0; $i <= $offsets_available; $i++ ) {
+	        $options[$i] = $i;
+        }
+
+        return $options;
+    }
+
+	/**
+	 * Display all section controls
+	 *
+	 * @param object  $section Current Section
+	 * @param array   $blocks  Our Sections Current Block
+	 * @param bool    $visible Show visible options?
+	 *
+	 * @since 1.2
+	 */
+	function mesh_section_controls( $section, $blocks, $visible = false ) {
+
+		$controls = array(
+			'visible_options' => array(
+				'template' => array(
+					'label'          => __( 'Columns', 'mesh' ),
+					'type'           => 'select',
+					'css_classes'    => array( 'mesh-choose-layout' ),
+					'validation_cb'  => false,
+                    'options_cb'     => array( $this, 'get_template_options' ),
+                    'id'             => 'mesh-sections-template-' . $section->ID,
+				),
+				'display-title' => array(
+					'label'          => __( 'Display Title', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-show-title' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+			),
+			'more_options' => array(
+				'css-class' => array(
+					'label'	         => __( 'CSS Class', 'mesh' ),
+					'type'           => 'text',
+					'css_classes'    => array( 'mesh-section-class' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+				'collapse' => array(
+					'label'          => __( 'Collapse Padding', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-collapse-input' ),
+					'show_on_cb'     => false,
+					'validation_cb'  => false,
+				),
+				'push-pull' => array(
+					'label'          => __( 'Push Pull', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => array( 'mesh-section-push' ),
+					'show_on_cb'     => array( $this, 'show_push_pull' ),
+					'validation_cb'  => false,
+				),
+				'featured_image' => array(
+					'type' => 'media',
+				),
+				'lp-equal' => array(
+					'label'          => __( 'Equalize', 'mesh' ),
+					'type'           => 'checkbox',
+					'css_classes'    => 'mesh-section-equalize',
+					'show_on_cb'     => array( $this, 'show_equalize' ),
+					'validate_cb'    => false,
+				),
+			),
+		);
+
+		$controls = apply_filters( 'mesh_section_controls', $controls );
+
+		if ( $visible ) {
+			$controls = $controls['visible_options'];
+		} else {
+			$controls = $controls['more_options'];
+		}
+
+		print( '<ul class="small-block-grid-1 medium-block-grid-4">' );
```

Comment:
pros / cons of using print? We use closing `?>`  do `<html stuff>` and reopen `<?php` in most scenarios (including below). So for consistency maybe we should replace?